qtacrylicmainwindow.cpp

```diff
@@ -0,0 +1,196 @@
+QtAcrylicMainWindow::~QtAcrylicMainWindow() = default;
+
+QColor QtAcrylicMainWindow::tintColor() const
+{
+    const QColor color = m_acrylicHelper.getTintColor();
+    if (color.isValid() && (color != Qt::transparent)) {
+        return color;
+    } else {
+        return palette().color(backgroundRole());
+    }
+}
+
+void QtAcrylicMainWindow::setTintColor(const QColor &value)
+{
+    if (!value.isValid()) {
+        qWarning() << "Tint color not valid.";
+        return;
+    }
+    if (m_acrylicHelper.getTintColor() != value) {
+        m_acrylicHelper.setTintColor(value);
+        QPalette pal = palette();
+        pal.setColor(backgroundRole(), m_acrylicHelper.getTintColor());
+        setPalette(pal);
+        //m_acrylicHelper.updateAcrylicBrush(tintColor());
+        update();
+        Q_EMIT tintColorChanged();
+    }
+}
+
+qreal QtAcrylicMainWindow::tintOpacity() const
+{
+    return m_acrylicHelper.getTintOpacity();
+}
+
+void QtAcrylicMainWindow::setTintOpacity(const qreal value)
+{
+    if (m_acrylicHelper.getTintOpacity() != value) {
+        m_acrylicHelper.setTintOpacity(value);
+        m_acrylicHelper.updateAcrylicBrush(tintColor());
+        update();
+        Q_EMIT tintOpacityChanged();
+    }
+}
+
+qreal QtAcrylicMainWindow::noiseOpacity() const
+{
+    return m_acrylicHelper.getNoiseOpacity();
+}
+
+void QtAcrylicMainWindow::setNoiseOpacity(const qreal value)
+{
+    if (m_acrylicHelper.getNoiseOpacity() != value) {
+        m_acrylicHelper.setNoiseOpacity(value);
+        m_acrylicHelper.updateAcrylicBrush(tintColor());
+        update();
+        Q_EMIT noiseOpacityChanged();
+    }
+}
+
+bool QtAcrylicMainWindow::frameVisible() const
+{
+    return m_frameVisible;
+}
+
+void QtAcrylicMainWindow::setFrameVisible(const bool value)
+{
+    if (m_frameVisible != value) {
+        m_frameVisible = value;
+        update();
+        Q_EMIT frameVisibleChanged();
+    }
+}
+
+QColor QtAcrylicMainWindow::frameColor() const
+{
+    return m_acrylicHelper.getFrameColor();
+}
+
+void QtAcrylicMainWindow::setFrameColor(const QColor &value)
+{
+    if (m_acrylicHelper.getFrameColor() != value) {
+        m_acrylicHelper.setFrameColor(value);
+        update();
+        Q_EMIT frameColorChanged();
+    }
+}
+
+qreal QtAcrylicMainWindow::frameThickness() const
+{
+    return m_acrylicHelper.getFrameThickness();
+}
+
+void QtAcrylicMainWindow::setFrameThickness(const qreal value)
+{
+    if (m_acrylicHelper.getFrameThickness() != value) {
+        m_acrylicHelper.setFrameThickness(value);
+        update();
+        Q_EMIT frameThicknessChanged();
+    }
+}
+
+bool QtAcrylicMainWindow::acrylicEnabled() const
+{
+    return m_acrylicOn;
+}
+
+void QtAcrylicMainWindow::setAcrylicEnabled(const bool value)
+{
+    m_acrylicOn = value;
+    if (m_acrylicOn) {
+        setAutoFillBackground(false);
+        setAttribute(Qt::WA_NoSystemBackground);
+        setAttribute(Qt::WA_OpaquePaintEvent);
+        setBackgroundRole(QPalette::Base);
```

Comment:
Can't these attributes be reset to default values if the user choose to turn off the acrylic blur?

Reply:
why would the default values be the right ones? I don't think toggling acrylic post startup is someone a user would do